modules/nextflow/src/main/groovy/nextflow/trace/AnsiLogObserver.groovy

```diff
@@ -371,34 +388,78 @@ class AnsiLogObserver implements TraceObserver {
     protected String fmtChop(String str, int cols) {
         if( str.size() <= cols )
             return str
-        return cols>3 ? str[0..(cols-3-1)] + '...' : str[0..cols-1]
+        return cols>5 ? str.take(3) + '…' + str.takeRight(cols-1-3) : str[0..cols-1]
     }
 
-    protected String line(ProgressRecord stats) {
+    private final static Pattern TAG_REGEX = ~/ \((.+)\)( *)$/
+    private final static Pattern LBL_REPLACE = ~/ \(.+\) *$/
+
+    protected Ansi line(ProgressRecord stats) {
+        final term = ansi()
         final float tot = stats.getTotalCount()
         final float com = stats.getCompletedCount()
         final label = fmtWidth(stats.taskName, labelWidth, Math.max(cols-50, 5))
+        final tagMatch = TAG_REGEX.matcher(label)
+        final labelTag = tagMatch ? tagMatch.group(1) : ''
+        final labelSpaces = tagMatch ? tagMatch.group(2) : ''
+        final labelNoTag = LBL_REPLACE.matcher(label).replaceFirst("")
+        final labelFinalProcess = labelNoTag.tokenize(':')[-1]
+        final labelNoFinalProcess = labelFinalProcess.length() > 0 ? labelNoTag - ~/$labelFinalProcess$/ : labelNoTag
```

Comment:
What `labelNoTag - ~/$labelFinalProcess$/` does? just strips `labelFinalProcess` from `labelNoTag`?

Reply:
Yup - I'm basically breaking that string into components so that I can colour them differently. I wanted to highlight the final process name (after the last `:`) as that's usually the most interesting bit.

Reply:
I added a chunk of comments a bit further up with an example, and what each variable will end up with.

Reply:
I think the regex is not necessary, have a look at feb0cc54